lib/routes/account.js

```diff
@@ -487,13 +489,20 @@ module.exports = function (
           if (shouldSendNewDeviceLoginEmail) {
             // The response doesn't have to wait for this,
             // so we don't return the promise.
-            mailer.sendNewDeviceLoginNotification(
-              emailRecord.email,
-              userAgent.call({
-                acceptLanguage: request.app.acceptLanguage,
-                timestamp: Date.now()
-              }, request.headers['user-agent'])
-            )
+            getGeoData(ip)
```

Comment:
I imagine since `getGeoData` returns a promise and can be asynchronous, there should be a `return` in front of this statement otherwise the next item in the chain can be called before this one completes.